integration/cronjobs_integration_test.go

```diff
@@ -0,0 +1,141 @@
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"connectrpc.com/connect"
+	"github.com/TBD54566975/ftl/backend/controller/cronjobs"
+	db "github.com/TBD54566975/ftl/backend/controller/dal"
+	"github.com/TBD54566975/ftl/backend/controller/scheduledtask"
+	"github.com/TBD54566975/ftl/backend/controller/sql/sqltest"
+	ftlv1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	"github.com/TBD54566975/ftl/backend/schema"
+	"github.com/TBD54566975/ftl/internal/cron"
+	"github.com/TBD54566975/ftl/internal/log"
+	"github.com/TBD54566975/ftl/internal/model"
+	"github.com/TBD54566975/ftl/internal/slices"
+	"github.com/alecthomas/assert/v2"
+	"github.com/alecthomas/types/optional"
+	"github.com/benbjohnson/clock"
+	"github.com/jpillora/backoff"
+)
+
+type mockScheduler struct {
+}
+
+func (s *mockScheduler) Singleton(retry backoff.Backoff, job scheduledtask.Job) {
+	// do nothing
+}
+
+func (s *mockScheduler) Parallel(retry backoff.Backoff, job scheduledtask.Job) {
+	// do nothing
+}
+
+type controller struct {
+	key      model.ControllerKey
+	dal      *db.DAL
+	clock    clock.Clock
+	cronJobs *cronjobs.Service
+}
+
+func TestServiceWithDB(t *testing.T) {
```

Comment:
Per my comment in the other PR, can we factor out the common code in the two tests such that they're using the same logic?